### PR TITLE
[circle2circle] Enable module optimization

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #4961 

Until now, only graph optimizations were possible.
This commit will enable module optimization in `circle2circle`.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/circle2circle/src/Circle2Circle.cpp [hunk 1]
```diff
@@ -290,11 +290,14 @@ int entry(int argc, char **argv)
...
+  // call luci optimizations for module
+  optimizer.optimize(module.get());
+
   for (size_t idx = 0; idx < module->size(); ++idx)
   {
     auto graph = module->graph(idx);
 
-    // call luci optimizations
+    // call luci optimizations for graph
     optimizer.optimize(graph);
```
Comment (mhs4670go) [inline compiler/circle2circle/src/Circle2Circle.cpp:301]: This could be removed. It being here is just for safety?

Comment (llFreetimell) [reply compiler/circle2circle/src/Circle2Circle.cpp:301]: `optimize(graph)` and `optimize(module)` is different and independent :)

For example, `FuseBCQPass` has some optimization for `module` but do not have for `graph`.
On the other hand, most of other passes has optimizations for `graph` but not for `module`.
Even worse, basic services such as `ShapeInferencePass` should be implemented for both of them.

Comment (mhs4670go) [reply compiler/circle2circle/src/Circle2Circle.cpp:301]: Oh, okay. I thought its functionality includes existing `optimize`'s one expecting `graph` as a parameter.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**